lib/matplotlib/cbook/__init__.py

```diff
@@ -2194,6 +2194,48 @@ def _check_in_list(_values, **kwargs):
                 .format(v, k, ', '.join(map(repr, values))))
 
 
+def _check_shape(_shape, **kwargs):
+    """
+    For each *key, value* pair in *kwargs*, check that *value* has the shape
+    *_shape*, if not, raise an appropriate ValueError.
+
+    None in the shape is treated as a "free" sizes that can have any length.
+    e.g. (None, 2) -> (N, 2)
+
+    The values checked must be numpy arrays.
+
+    Examples
+    --------
+    To check from (N, 2) shaped arrays
+
+    >>> cbook._check_in_list((None, 2), arg=arg, other_arg=other_arg)
+    """
+    target_shape = _shape
+    for k, v in kwargs.items():
+        data_shape = v.shape
+
+        if not (
+            len(data_shape) == len(target_shape)
+            and all(
+                (t == s if t is not None else True)
+                for t, s in zip(target_shape, data_shape)
+            )
+        ):
```

Comment:
```
        if (len(data_shape) != len(target_shape)
                or any(t not in [d, None]
                       for d, t in zip(data_shape, target_shape))):
```
(not so much the formatting (heh :-)), just the inverted logic seems easier to read?)

Reply:
I had it that way eventually and thought the `and` was clearer than the `or`.  the `t not in [d, None]` is much nicer than the ternary though.